app/upgrades/vesting.go

```diff
@@ -0,0 +1,273 @@
+package upgrades
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authvestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+
+	"github.com/CosmosContracts/juno/v16/app/keepers"
+)
+
+const (
+	// TODO: Ensure mainnet codeId is used here
+	// Same as Reece, Noah, and Ekez contracts.
+	// junod q wasm code 2453 $HOME/Desktop/vesting.wasm --node https://juno-rpc.reece.sh:443
+	vestingCodeID = 2453
+	// vestingCodeID        = 1 // testing
+	junoUnbondingSeconds = 2419200
+)
+
+type VestingContract struct {
+	Owner                    string `json:"owner"`
+	Recipient                string `json:"recipient"`
+	Title                    string `json:"title"`
+	Description              string `json:"description"`
+	Schedule                 string `json:"schedule"`
+	UnbondingDurationSeconds uint64 `json:"unbonding_duration_seconds"`
+	VestingDurationSeconds   uint64 `json:"vesting_duration_seconds"`
+	Total                    string `json:"total"`
+	Denom                    Denom  `json:"denom"`
+}
+
+type Denom struct {
+	Native string `json:"native"`
+}
+
+// Stops a vesting account and returns all tokens back to the Core-1 SubDAO.
+func MoveVestingCoinFromVestingAccount(ctx sdk.Context, keepers *keepers.AppKeepers, bondDenom string, name string, accAddr sdk.AccAddress, core1AccAddr sdk.AccAddress) error {
+	now := ctx.BlockHeader().Time
+
+	stdAcc := keepers.AccountKeeper.GetAccount(ctx, accAddr)
+	vacc, ok := stdAcc.(*authvestingtypes.PeriodicVestingAccount)
+	if !ok {
+		// For e2e testing
+		fmt.Printf("account " + accAddr.String() + " is not a vesting account.\n")
+		return nil
+	}
+
+	fmt.Printf("\n\n== Vesting Account Address: %s (%s) ==\n", vacc.GetAddress().String(), name)
+
+	// Gets non-vested coins (These get returned back to Core-1 SubDAO / a new vesting contract made from)
+	unvestedCoins := vacc.GetVestingCoins(now)
+	fmt.Printf("Locked / waiting to vest Coins: %v\n", unvestedCoins)
+
+	// Pre migration balance
+	core1BeforeBal := keepers.BankKeeper.GetBalance(ctx, core1AccAddr, bondDenom)
+
+	// Clears the account so all all future vesting periods are removed.
+	clearVestingAccount(ctx, vacc, keepers, unvestedCoins)
+
+	// Complete any re-deleations to become standard delegations.
+	if err := completeAllRedelegations(ctx, keepers, accAddr, now); err != nil {
+		return err
+	}
+
+	// Instant unbond all delegations.
+	_, err := unbondAllAndFinish(ctx, now, keepers, accAddr)
```

Comment:
Why do we need to undelegate everything?

Reply:
tldr; hassle and minimizing risk

1) I clear the vesting account so that all vesting periods are unlocked. If a Core-1 member has more than 51% of vested tokens delegated, it's a nightmare to do the math. 
This reduces risk of incorrect math and just ensures all tokens are undone before trying to bankSend -> the Core-1 DAO -> vesting contract init.

2) All these vested tokens are moving to the new vesting contracts so we have to unvest them. I can't move delegations to the new contracts until they are created, which can only be done after I unvest all funds and return them to the base account.

Then Core-1 members can re-stake after the upgrade in the new CW vesting UI
_(I will get the withdraw/recipient addresses from you guys in the coming days)_

**NOTE**: Any rewards / vested funds which a Core-1 member has in their SDK vesting account will STAY in said account since they are the controller of it / it has been unlocked.